iothub_client/src/blob.c

```diff
@@ -131,257 +135,263 @@ BLOB_RESULT Blob_UploadBlock(
+            LogError("hostname cannot be determined");
+            result = BLOB_INVALID_ARG;
+        }
+        else
+        {
+            size_t hostnameSize = relativePath - hostnameBegin;
+            if ((hostname = (char*)malloc(hostnameSize + 1)) == NULL)
+            {
+                /*Codes_SRS_BLOB_02_016: [ If the hostname copy cannot be made then then Blob_UploadMultipleBlocksFromSasUri shall fail and return BLOB_ERROR ]*/
+                LogError("oom - out of memory");
+                result = BLOB_ERROR;
+            }
+            else
+            {
+                (void)memcpy(hostname, hostnameBegin, hostnameSize);
+                hostname[hostnameSize] = '\0';
 
-                            }
-                            HTTPAPIEX_Destroy(httpApiExHandle);
-                        }
-                        free(hostname);
-                    }
+                /*Codes_SRS_BLOB_02_018: [ Blob_UploadMultipleBlocksFromSasUri shall create a new HTTPAPI_EX_HANDLE by calling HTTPAPIEX_Create passing the hostname. ]*/
+                httpApiExHandle = HTTPAPIEX_Create(hostname);
+                if (httpApiExHandle == NULL)
+                {
+                    /*Codes_SRS_BLOB_02_007: [ If HTTPAPIEX_Create fails then Blob_UploadMultipleBlocksFromSasUri shall fail and return BLOB_ERROR. ]*/
+                    LogError("unable to create a HTTPAPIEX_HANDLE");
+                    result = BLOB_ERROR;
+                }
+                else if ((certificates != NULL) && (HTTPAPIEX_SetOption(httpApiExHandle, "TrustedCerts", certificates) == HTTPAPIEX_ERROR))
+                {
+                    LogError("failure in setting trusted certificates");
+                    result = BLOB_ERROR;
+                }
+                else if ((proxyOptions != NULL && proxyOptions->host_address != NULL) && HTTPAPIEX_SetOption(httpApiExHandle, OPTION_HTTP_PROXY, proxyOptions) == HTTPAPIEX_ERROR)
+                {
+                    LogError("failure in setting proxy options");
+                    result = BLOB_ERROR;
+                }
+                /*Codes_SRS_BLOB_02_028: [ Blob_UploadMultipleBlocksFromSasUri shall construct an XML string with the following content: ]*/
+                else if ((blockIDList = STRING_construct(blockListXmlBegin)) == NULL)
+                {
+                    /*Codes_SRS_BLOB_02_033: [ If any previous operation that doesn't have an explicit failure description fails then Blob_UploadMultipleBlocksFromSasUri shall fail and return BLOB_ERROR ]*/
+                    LogError("failed to STRING_construct");
+                    result = BLOB_HTTP_ERROR;
+                }
+                else if ((result = InvokeUserCallbackAndSendBlobs(httpApiExHandle, relativePath, blockIDList, getDataCallbackEx, context, httpStatus, httpResponse)) != BLOB_OK)
+                {
+                   ; 
```

Comment:
We should have a log trace here for the error. 
We need to reach the million-log-trace mark every time we run a simple sample, otherwise we fall below SLA.

Reply:
Fixed.